starknet_py/net/full_node_client.py

```diff
@@ -536,16 +574,34 @@ def _create_broadcasted_txn(transaction: AccountTransaction) -> dict:
     }
 
 
-def _create_broadcasted_declare_properties(transaction: Declare) -> dict:
-    contract_class = cast(Dict, DeclareSchema().dump(obj=transaction))["contract_class"]
+def _create_broadcasted_declare_properties(
+    transaction: Union[Declare, DeclareV2]
+) -> dict:
+    contract_class = (
+        cast(Dict, DeclareSchema().dump(obj=transaction))["contract_class"]
+        if isinstance(transaction, Declare)
+        else cast(Dict, DeclareV2Schema().dump(obj=transaction))["contract_class"]
+    )
     declare_properties = {
         "contract_class": {
-            "program": contract_class["program"],
             "entry_points_by_type": contract_class["entry_points_by_type"],
             "abi": contract_class["abi"],
         },
         "sender_address": _to_rpc_felt(transaction.sender_address),
     }
+    if isinstance(transaction, DeclareV2):
+        declare_properties["contract_class"]["sierra_program"] = contract_class[
+            "sierra_program"
+        ]
+        declare_properties["contract_class"]["contract_class_version"] = contract_class[
+            "contract_class_version"
+        ]
+        declare_properties["compiled_class_hash"] = _to_rpc_felt(
+            transaction.compiled_class_hash
+        )
+    else:
+        declare_properties["contract_class"]["program"] = contract_class["program"]
+
     return declare_properties
```

Comment:
Could we have `_create_broadcasted_declare_v2_properties`? A lot is happening in this function